### PR TITLE
Update UserGuide and DeveloperGuide

### LINKED ISSUE #213: Complete UserGuide for 1.3 Additions

### PR DESCRIPTION
**Pull request by lye-jw**
Resolves #213

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -194,33 +195,50 @@ image::delete/Step3.png[,550]
...
-*   `update i/1 n/Asam Laksa v/11 r/My first time trying! t/Dinner t/Food`
+* All fields are optional, but you must specify at least one field to be updated.
 
-Expected output: +
+* You can update multiple fields at once.
+****
 
-NOTE: Assume 2nd command occurs after 1st
+==== Example Usage:
+You order a bowl of delicious Laksa and key in the details of this food expense into THRIFT. However, after your meal you realize you recorded some incorrect details - the Laksa was _Asam Laksa_, not Curry Laksa as you thought, and you paid more than the menu price (which you recorded) due to taxes. You also want to add a remark and tag.
```
Comment (limerencee) [inline docs/UserGuide.adoc:219]: You ordered .. and keyed in ...
... not Curry Laksa. You also paid more than the menu price due to taxes.

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -194,33 +195,50 @@ image::delete/Step3.png[,550]
...
 
-*   `Updated Transaction: [-] Curry Laksa ($10.50) Date: 12/10/2019 Remarks:  Tags: [Lunch][Food]` +
-+
-`Original: [-] Laksa ($10.50) Date: 12/10/2019 Remarks:  Tags: [Lunch][Food]`
+Fortunately, you can easily update such details in THRIFT. Here is what you do from beginning to end:
 
-*   `Updated Transaction: [-] Asam Laksa ($11.00) Date: 12/10/2019 Remarks: My first time trying! Tags: [Dinner][Food]` +
+. You initially have a "Curry Laksa" expense in THRIFT. It is at index 2 in the list.
++
+image::update/updateUG/UpdateStep1.png[]
```
Comment (limerencee) [inline docs/UserGuide.adoc]: Should stick specify the size of the image as [, 550]

File: docs/UserGuide.adoc [hunk 3]
```diff
@@ -310,23 +328,51 @@ The convert command allows you to convert money from one currency to another.
...
-Format: `clone i/INDEX`
+==== Command syntax:
+`clone i/INDEX [o/OCCURRENCE]`
+
+The following is an explanation for the `clone` command:
 
 ****
-* Clones (creates a duplicate of) a `Transaction` entry.
-* Clone produced is added to a new index at the bottom of the list.
+* Clones (creates duplicate(s) of) transaction at `INDEX`. Clone(s) produced are added to a new index at the bottom of the list.
```
Comment (limerencee) [inline docs/UserGuide.adoc:341]: Clones (creates duplicate(s)) of transaction ...

File: docs/UserGuide.adoc [hunk 4]
```diff
@@ -310,23 +328,51 @@ The convert command allows you to convert money from one currency to another.
...
+
+- `INDEX` refers to the index currently shown in the displayed list, and must be a positive integer.
+
+* If you use `clone i/INDEX` without `o/OCCURRENCE` field, one clone is created for the current day.
+
+* Optional `OCCURRENCE` field is in the format `FREQUENCY:NUMBER_OF_OCCURRENCES`. You use it to create multiple clones,  with the 1st clone on the current day.
+
+- `FREQUENCY` is how often or how far apart the clones occur. Valid values are `daily`, `weekly`, `monthly` and `yearly`.
```
Comment (limerencee) [inline docs/UserGuide.adoc]: Valid values are:

**Review by limerencee — CHANGES_REQUESTED** [review]
LGTM after minor changes

File: docs/UserGuide.adoc (hunk 1 above)
Comment (lye-jw) [reply docs/UserGuide.adoc:219]: I believe the present tense is fine as it is a continuous story from start to end. Each action is taken at a step in the present.

File: docs/UserGuide.adoc (hunk 3 above)
Comment (lye-jw) [reply docs/UserGuide.adoc:341]: When separated it is "Clones transaction" and "creates duplicate(s) of transaction" so this is correct.

File: docs/UserGuide.adoc (hunk 4 above)
Comment (lye-jw) [reply docs/UserGuide.adoc]: Good catch!

**Review by limerencee — APPROVED** [review]

**Pull request merged**